biosimulators_utils/model_lang/smoldyn/utils.py

```diff
@@ -6,18 +6,65 @@
 :License: MIT
 """
 
+
 from ...config import Config  # noqa: F401
 from ...sedml.data_model import (SedDocument, ModelAttributeChange, Variable,  # noqa: F401
                                  Symbol, Simulation, UniformTimeCourseSimulation, Algorithm,
                                  Task)
 from ...utils.core import flatten_nested_list_of_strings
-from .validation import validate_model
+from .validation import generate_model_validation_object, validate_model
 from smoldyn.biosimulators.utils import read_simulation
+from ..smoldyn.simularium_converter import SmoldynDataConverter, CombineArchive
 import os
 import re
 import types  # noqa: F401
+from typing import Optional  # noqa: F401
+
+
+__all__ = [
+    'get_parameters_variables_outputs_for_simulation',
+    'generate_new_simularium_file',
+]
+
+
+def generate_new_simularium_file(archive_rootpath: str, simularium_filename: str, save_output_df: bool = False) -> None:
+    """Generate a new `.simularium` file based on the `model.txt` in the passed-archive rootpath using the above
+        validation method. Raises an `Exception` if there are errors present.
 
-__all__ = ['get_parameters_variables_outputs_for_simulation']
+    Args:
+        archive_rootpath (:obj:`str`): Parent dirpath relative to the model.txt file.
+        simularium_filename (:obj:`str`): Desired save name for the simularium file to be saved
+            in the `archive_rootpath`.
+        save_output_df (:obj:`bool`): Whether to save the modelout.txt contents as a pandas df in csv form. Defaults
+            to `False`.
+
+    Returns:
+        None
+    """
+    archive = CombineArchive(rootpath=archive_rootpath, name=simularium_filename)
+    model_validation = generate_model_validation_object(archive)
+    if model_validation.errors:
+        print(f'There are errors involving your model file:\n{model_validation.errors}\nPlease adjust your model file.')
+        raise ValueError
```

Comment:
add printed message to the ValueError so that caller has some idea what is going on.  also, logs are better than prints (formatted and you can turn them off).

Reply:
That was yet another placeholder in line for a log.